promutils/labeled/counter.go

```diff
@@ -59,8 +61,16 @@ func NewCounter(name, description string, scope promutils.Scope, opts ...MetricO
 		if _, emitUnlabeledMetric := opt.(EmitUnlabeledMetricOption); emitUnlabeledMetric {
 			c.Counter = scope.MustNewCounter(GetUnlabeledMetricName(name), description)
 		} else if additionalLabels, casted := opt.(AdditionalLabelsOption); casted {
-			c.CounterVec = scope.MustNewCounterVec(name, description, append(metricStringKeys, additionalLabels.Labels...)...)
-			c.additionalLabels = contextutils.MetricKeysFromStrings(additionalLabels.Labels)
+			labels := make([]string, 0, len(additionalLabels.Labels))
+			metricKeysSet := sets.NewString(metricStringKeys...)
+			for _, label := range additionalLabels.Labels {
+				if !metricKeysSet.Has(label) {
+					labels = append(labels, label)
+				}
+			}
+			// Here we only append the labels that don't exist in metricStringKeys
+			c.CounterVec = scope.MustNewCounterVec(name, description, append(metricStringKeys, labels...)...)
+			c.additionalLabels = contextutils.MetricKeysFromStrings(labels)
```

Comment:
Should we apply the same logic to other metrics under labeled package?

Reply:
yeah, updated it